src/mon/MDSMonitor.cc

```diff
@@ -1608,6 +1608,19 @@ int MDSMonitor::management_command(
       return -EINVAL;
     }
 
+    for (auto fs : pending_fsmap.get_filesystems()) {
+      const set<int64_t>& data_pools = fs.second->mds_map.get_data_pools();
+      string sure;
+      if ((data_pools.find(data) != data_pools.end()
+	   || fs.second->mds_map.metadata_pool == metadata)
+	  && ((!cmd_getval(g_ceph_context, cmdmap, "allow_overlay", sure)
+	       || sure != "--allow-dangerous-metadata-overlay"))) {
```

Comment:
My original thought was that we would only permit this if one of the filesystems is marked damaged, so that there was no way for the user to have two running MDSs pointing at the same pool.  We should probably still do that, or at least enforce that cluster_down is set for one of the filesystems at all times -- users are careless with the --this-is-so-dangerous-i-cant-even flags.